Tests/DequeTests/DequeTests.swift

```diff
@@ -374,5 +374,40 @@ final class DequeTests: CollectionTestCase {
     }
   }
 
+  func test_shrinkCapacityWhenEmpty() {
+    withEvery("capacity", in: [0, 1, 2, 5, 10, 50, 100]) { capacity in
+      var deque = Deque<Int>(minimumCapacity: capacity)
+
+      XCTAssertGreaterThanOrEqual(deque._capacity, capacity)
```

Comment:
Nit: Please use the custom `expect*` assertions from _CollectionsTestSupport -- `XCTAssert*` does not know about the context, so its failure messages will not identify the case that failed, making debugging much harder.